assets/js/atomic/blocks/product-elements/button/block.js

```diff
@@ -147,12 +160,7 @@ const AddToCartButton = ( {
 			dispatchStoreEvent( 'cart-add-item', {
```

Comment:
I guess that the problem here is that `addToCart` returns a Promise and we are not waiting for it.

I commented the new code, and edited the function `onClick`, in this way:

```
		buttonProps.onClick = async () => {
			await addToCart();
			dispatchStoreEvent( 'cart-add-item', {
				product,
			} );
			const { cartRedirectAfterAdd } = getSetting( 'productsSettings' );
			if ( cartRedirectAfterAdd ) {
				window.location.href = CART_URL;
			}
			setClickedAddToCart( true );
		};
	} 
```

It seems that works. Can you check?

Reply:
It's a great solution! Thanks so much! I updated the PR per your suggestion!